Create official Documentation (#11)
flagbit/category-bundle: docs/custom-types.md

## Create custom property types

> While the Category Bundle comes with many build-in property types, your project might need one that is too specific to
> your use case than it could be added to the Category Bundle. Don't worry, it allows you to create your own property type
> that can be used in your project without contributing to the Bundle.

Reviewer: Maybe we should rephrase this a bit, as the first sentence is hard to read.

Maybe something like?
```suggestion
While the Category Bundle comes with many build-in property types, your project might need one that is very specific to your use case. Adding such a specific type to the official bundle might not make sense. Don't worry, the bundle allows you to create your own property type that can be used in your project without contributing to the Bundle.
```

your bundle:

``` typescript
// Resources/public/js/property/my-property.tsx
import * as React from 'react';
import { Property, PropertyFactory } from './property';
import PropertyDto from './property-dto';

class MyProperty implements Property {
    render(propertyDto: PropertyDto): React.ReactNode {
        return (
            <React.Fragment key={propertyDto.code + propertyDto.locale}>
                <div className="AknFieldContainer-inputContainer field-input">
                    <input
                        id={propertyDto.createId()}
                        type={'text'}
                        value={propertyDto.value}
                        className={'AknTextField'}
                        onChange={(event: React.ChangeEvent<HTMLInputElement>): void => {
                            propertyDto.updateValue(event.target.value);
                        }}
                    />
                </div>
            </React.Fragment>
        );
    }
}

const factory: PropertyFactory = (): Property => new MyProperty();

export default factory;
```

The implemented `render()` will get an instance of `PropertyDto` which also contains a method to register changes that

> occur when you do changes in your custom property on the category page. You can also access:

Reviewer: This sentence is also really complex, consider rephrasing it.

Author: @pascal-zarrad What do you think about:

```suggestion
The implemented `render()` method will get an instance of `PropertyDto` injected. The `PropertyDto` contains a method to 
apply changes from your custom property on the category page. You can also access:
```

Reviewer: @flagbird this would be fine 👍🏻
